### PR TITLE
bump pt to 2.5.1

### PR DESCRIPTION
**Pull request by t-vi**
Fixes: #1372

based on the discussion in #1372

- pypi.python.org is actually the right repo for nvfuser built for stable PT,
- as of now, we have -torch25 packages there and the plan is to update these regularly (and it's not so clear with -toirch24).

### CODE REVIEW

File: dockers/ubuntu-cuda/Dockerfile [hunk 1]
```diff
@@ -160,7 +160,7 @@ RUN \
         # installing pytorch from wheels \
         CUDA_VERSION_MM=${CUDA_VERSION%.*} && \
         TORCH_VERSION_MM=${TORCH_VERSION%.*} && \
-        pip install -U "nvfuser-cu${CUDA_VERSION_MM/./}-torch${TORCH_VERSION_MM/./}" ; \
+        pip install -U --extra-index-url https://pypi.nvidia.com --pre "nvfuser-cu${CUDA_VERSION_MM/./}-torch${TORCH_VERSION_MM/./}" ; \
```
Comment (Borda) [inline dockers/ubuntu-cuda/Dockerfile:163]:
```suggestion
        pip install -U --extra-index-url https://pypi.nvidia.com "nvfuser-cu${CUDA_VERSION_MM/./}-torch${TORCH_VERSION_MM/./}" ; \
```
Is the Pre needed? Does this mean you are using unstable versions?

File: .azure/docker-build.yml [hunk 2]
```diff
@@ -110,6 +110,6 @@ jobs:
           echo $(imageRepository):$(imageTag)
           echo $(DOCKERHUB_PAT) | docker login --username $(DOCKERHUB_USER) --password-stdin
           docker push $(imageRepository):$(imageTag)-dev
-        condition: ne(variables['Build.Reason'], 'PullRequest')
+        #condition: ne(variables['Build.Reason'], 'PullRequest')
```
Comment (Borda) [inline .azure/docker-build.yml:113]:
```suggestion
        condition: ne(variables['Build.Reason'], 'PullRequest')
```

**Review by Borda — APPROVED** [review]
`nvfuser_cu121_torch24    0.2.9.dev20240808`

File: dockers/ubuntu-cuda/Dockerfile (hunk 1 above)
Comment (t-vi) [reply dockers/ubuntu-cuda/Dockerfile]: nvfuser only has pre-releases.

**Pull request merged**